### PR TITLE
Head Decap VR

### PR DESCRIPTION
**Pull request by mohsenheydari**
## Summary

Features:

- Avatar moves/rotates with HMD
- Camera movement sync with controller's input
- Fixes XR inputs transform issues
- Fixes XR input network sync
- Adds default controller grip mesh to non-local clients

## Checklist
- [ ] Pre-push checks pass `npm run check`
  - [ ] Linter passing via `npm run lint`
  - [ ] Unit & Integration tests passing via `npm run test:packages`
  - [ ] Docker build process passing via `npm run build-client`
- [ ] If this PR is still a WIP, convert to a draft 
- [ ] When this PR is ready, mark it as "Ready for review"
- [ ] Changes have been manually QA'd
- [ ] Changes reviewed by at least 2 approved reviewers

## References

References to pertaining issue(s)

## QA Steps

1. `git checkout pr_branch_name`
2. `npm install`
3. `npm run dev-reinit`
4. `npm run dev`

List any additional steps required to QA the changes of this PR, as well as any supplemental images or videos.

## Reviewers

@HexaField @speigg

### CODE REVIEW

File: packages/engine/src/xr/functions/WebXRFunctions.ts [hunk 1]
```diff
@@ -147,15 +133,11 @@ export const startWebXR = (): void => {
...
   // Sometimes the input sources are not available immidiately
-  setTimeout(() => {
-    mapXRControllers(inputData)
-    proxifyXRInputs(world.localClientEntity, inputData)
-  }, 1000)
-
-  addComponent(world.localClientEntity, XRInputSourceComponent, inputData)
-
-  bindXRHandEvents()
+  await delay(1000)
```
Comment (HexaField) [inline packages/engine/src/xr/functions/WebXRFunctions.ts:136]: can we use a threejs or WebXR API event instead of a delay?

**Review by HexaField — CHANGES_REQUESTED** [review]

File: packages/engine/src/xr/functions/WebXRFunctions.ts (hunk 1 above)
Comment (mohsenheydari) [reply packages/engine/src/xr/functions/WebXRFunctions.ts:136]: For some reason, the 'connect' event from three.js does not trigger the first time going into immersive mode. That's why I have used delay. I'll try to find an alternative solution.

Comment (mohsenheydari) [reply packages/engine/src/xr/functions/WebXRFunctions.ts]: Replaced the delay with 'inputsourceschange' event from XRSession. Testing on the emulator and Quest 2 seem to be working without issues.

File: packages/engine/src/avatar/AvatarControllerSystem.ts [hunk 2]
```diff
@@ -35,24 +48,39 @@ export default async function AvatarControllerSystem(world: World) {
     }
 
     for (const entity of localXRInputQuery.enter(world)) {
-      const avatar = getComponent(entity, AvatarComponent)
+      // TODO: The opacity value does not set immidiately
+      setTimeout(() => setAvatarHeadOpacity(entity, 0), 1000)
```
[the remark this answers is not in the document]
Comment (mohsenheydari) [reply packages/engine/src/avatar/AvatarControllerSystem.ts]: Removed the 'setTimeout's for now

File: packages/engine/src/avatar/functions/avatarFunctions.ts [hunk 3]
```diff
@@ -274,3 +274,28 @@ const addBoneOpacityParamsToMaterial = (material, boneIndex = -1) => {
...
+  object3DComponent?.value.traverse((obj) => {
+    if (!(obj as SkinnedMesh).isSkinnedMesh) return
+    const material = (obj as SkinnedMesh).material as Material
+    if (!material.userData.shader) return
+    const shader = material.userData.shader
+    shader.uniforms.boneOpacity.value = opacity
+  })
+}
+
+export const getAvatarBoneWorldPosition = (entity: Entity, name: string, position: Vector3): void => {
```
[the remark this answers is not in the document]
Comment (mohsenheydari) [reply packages/engine/src/avatar/functions/avatarFunctions.ts]: Replaced the search with `IKRigComponent.boneStructure`

**Review by HexaField — APPROVED** [review]

**Pull request merged**